src/service/AggregatedTransactionService.ts

```diff
@@ -0,0 +1,127 @@
+     * @param signedTransaction - The signed aggregate transaction (complete) to be verified
+     * @returns {Observable<boolean>}
+     */
+    public isComplete(signedTransaction: SignedTransaction): Observable<boolean> {
+        const aggregateTransaction = TransactionMapping.createFromPayload(signedTransaction.payload) as AggregateTransaction;
+        /**
+         * Include both initiator & cosigners
+         */
+        const signers = (aggregateTransaction.cosignatures.map((cosigner) => cosigner.signer.publicKey));
+        if (signedTransaction.signer) {
+            signers.push(signedTransaction.signer);
+        }
+
+        return observableFrom(aggregateTransaction.innerTransactions).pipe(
+            mergeMap((innerTransaction) => this.accountHttp.getMultisigAccountInfo(innerTransaction.signer.address)
+                .pipe(
+                    /**
+                     * For multisig account, we need to get the graph info in case it has multiple levels
+                     */
+                    mergeMap((_) => _.minApproval !== 0 && _.minRemoval !== 0 ?
+                        this.accountHttp.getMultisigAccountGraphInfo(_.account.address)
+                        .pipe(
+                            map((graphInfo) => this.validateCosignatories(graphInfo, signers)),
+                        ) : observableOf(true),
+                        ),
+                    ),
+                ),
+            );
+    }
+
+    /**
+     * Validate cosignatories against multisig Account(s)
+     * @param graphInfo - multisig account graph info
+     * @param cosignatories - array of cosignatories extracted from aggregated transaction
+     * @returns {boolean}
+     */
+    private validateCosignatories(graphInfo: MultisigAccountGraphInfo, cosignatories: string[]): boolean {
+        /**
+         *  Validate cosignatories from bottom level to top
+         */
+        const sortedKeys = Array.from(graphInfo.multisigAccounts.keys()).sort((a, b) => b - a);
+        const cosignatoriesReceived = cosignatories;
+        let validationResult = true;
+
+        sortedKeys.forEach((key) => {
+            const multisigInfo = graphInfo.multisigAccounts.get(key);
+            if (multisigInfo && validationResult) {
+                multisigInfo.forEach((multisig) => {
+                    if (multisig.minApproval >= 1) {
+                        const matchedCosignatories = this.compareArrays(cosignatoriesReceived,
+                                        multisig.cosignatories.map((cosig) => cosig.publicKey));
+
+                        /**
+                         * if minimal signature requirement met at current level, push the multisig account
+                         * into the received signatories array for next level validation.
+                         * Otherwise return validation failed.
+                         */
+                        if (matchedCosignatories.length >= multisig.minApproval) {
```

Comment:
Will this method `validateCosignatories` also be called in case of **removal** of cosignatories with `MultisigModificationTransaction` ? 

If this is the case, this line will cause problems because `minApproval` and `minRemoval` can be different. And they both define the `minimum count of needed cosignatories` but in different cases. Probably this line would need to make the different of which field is needed..

Please discuss

Reply:
done 👍